src/rp2040/serial.c

```diff
@@ -67,10 +104,17 @@ serial_enable_tx_irq(void)
 void
 serial_init(void)
 {
+
+#if CONFIG_RP2040_SERIAL_UART0
     enable_pclock(RESETS_RESET_UART0_BITS);
+    uint32_t pclk = get_pclock_frequency(RESETS_RESET_UART0_BITS);
+#elif CONFIG_RP2040_SERIAL_UART1
+    enable_pclock(RESETS_RESET_UART1_BITS);
+    uint32_t pclk = get_pclock_frequency(RESETS_RESET_UART1_BITS);
+#endif
```

Comment:
Please avoid `#if` in the executable code - prefer `if (CONFIG_RP2040_SERIAL_UART0) ...`